actions/views/channel.js

```diff
@@ -95,7 +95,9 @@ export function leaveChannel(channelId) {
         }
 
         const teamUrl = getCurrentRelativeTeamUrl(state);
-        browserHistory.push(teamUrl + '/channels/' + Constants.DEFAULT_CHANNEL);
+        const currentTeamId = getCurrentTeamId(state);
+        const redirectChannel = getRedirectChannelNameForTeam(state, currentTeamId);
+        browserHistory.push(teamUrl + '/channels/' + redirectChannel);
```

Comment:
Similar to my other comment, would it be possible to just switch to `teamUrl` and have the redirect logic figure out the redirect channel?

It might be possible to make it so that the ChannelIndentifierRouter is the only place that needs to know about the default and redirect channel, and it would make it easier to change that logic in the future. I think we discussed making a change like that previously, and it might be a good time to do that.

That being said, that's a big refactor, so I can totally understand not doing that right now.